src/transformers/modeling_tapas.py

```diff
@@ -0,0 +1,1408 @@
+# coding=utf-8
+# Copyright (...) and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PyTorch TAPAS model. """
+
+
+import math
+import os
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+from torch.nn import CrossEntropyLoss, MSELoss
+
+from .activations import ACT2FN
+from .configuration_tapas import TapasConfig
+from .file_utils import (ModelOutput, 
+                        add_start_docstrings, 
+                        add_start_docstrings_to_callable,
+                        replace_return_docstrings,
+                        is_scatter_available,
+                        requires_scatter,
+)
+from .modeling_outputs import (
+    BaseModelOutput, 
+    BaseModelOutputWithPooling, 
+    MaskedLMOutput,
+    SequenceClassifierOutput,
+)
+from .modeling_utils import (
+    PreTrainedModel,
+    apply_chunking_to_forward,
+    find_pruneable_heads_and_indices,
+    prune_linear_layer,
+)
+from .utils import logging
+
+# soft dependency
+if is_scatter_available():
+    from transformers import modeling_tapas_utilities as utils
+
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "TapasConfig"
+_TOKENIZER_FOR_DOC = "TapasTokenizer"
+
+TAPAS_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "tapas-base",
+    "tapas-large",
+    # See all TAPAS models at https://huggingface.co/models?filter=tapas
+]
```

Comment:
Yes, some notes about this:

* I would like to discuss which of the models to upload to the model hub, because the authors did release a large amount of model variants: https://github.com/google-research/tapas. I assume not all of them are that useful, especially the TINY models, and maybe we can only include the ones with relative embeddings (because performance is almost always slightly better with relative position embeddings) and intermediate pre-training (for the same reason). 

* The naming of the models is quite extensive: **tapas_wtq_wikisql_sqa_inter_masklm_large_reset** for example is the large model with relative position embeddings (reset) that was pretrained on masked language modeling, then intermediate pretraining, then fine-tuned on SQA, then on WikiSQL and finally on WTQ 😅 Is there a naming standard for models in the model hub? Maybe this can be shortened to **tapas-large-finetuned-wtq**, and provide the details in the model cards.

Reply:
- Regarding the models to be added, I don't think it hurts to put all the models in the hub. This way it would be easy for future researchers to replicate the results of this paper, only using the implementation here, without needing to switch from one implementation to another. In any case, the conversion script should handle all of these.  We can take care of uploading the models then, as they'll need to be under the google organization namespace.

- I think your proposal regarding model naming makes total sense! As long as there exists a document in which we can find the old name, renaming to something easier to understand like this is fine.